src/main/java/com/npe/pet/uncaughtHeroes/controller/HeroController.java

```diff
@@ -0,0 +1,56 @@
+package com.npe.pet.uncaughtHeroes.controller;
+
+import com.npe.pet.uncaughtHeroes.entity.Hero;
+import com.npe.pet.uncaughtHeroes.service.HeroService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/heroes")
+public class HeroController {
+
+    private final HeroService heroService;
+
+    @Autowired
```

Comment:
I believe, that the @Autowired annotation is not needed here.

You can simply delete it from the constructor, and then Spring will do dependency injection and that way it will work fine.

Also, you could remove the constructor, and remove the final keyword in the declaration of the variable and annotate the declaration with @Autowired. I think this would also work fine.

Reply:
Good catch!
The base of this class was generated by ChatGPT and I didn't notice this. The HeroService class also has the same, probably for similar reason. 😄 I prefer having a constructor and no @Autowired annotations, so I will remove it. (That is how I used it in the TestDataInitializer component)